dask/array/core.py

```diff
@@ -3042,101 +3067,153 @@ def normalize_chunks(chunks, shape=None, limit=None, dtype=None, previous_chunks
 
     Respects null dimensions
 
+    >>> normalize_chunks(())
+    ()
+    >>> normalize_chunks((), ())
+    ()
     >>> normalize_chunks((), shape=(0, 0))
     ((0,), (0,))
+
+    Handles NaNs
+
+    >>> normalize_chunks((1, (np.nan,)), (1, np.nan))
+    ((1,), (nan,))
     """
-    if dtype and not isinstance(dtype, np.dtype):
-        dtype = np.dtype(dtype)
-    if chunks is None:
-        raise ValueError(CHUNKS_NONE_ERROR_MESSAGE)
-    if isinstance(chunks, list):
-        chunks = tuple(chunks)
-    if isinstance(chunks, (Number, str)):
-        chunks = (chunks,) * len(shape)
-    if isinstance(chunks, dict):
-        chunks = tuple(chunks.get(i, None) for i in range(len(shape)))
-    if isinstance(chunks, np.ndarray):
-        chunks = chunks.tolist()
-    if not chunks and shape and all(s == 0 for s in shape):
-        chunks = ((0,),) * len(shape)
+    if shape is not None:
+        shape_: tuple[T_IntOrNaN, ...] = shape
+        shape_len: int = len(shape)
+        shape_is_not_none: bool = True
+    else:
+        shape_ = ()
+        shape_len = -1
+        shape_is_not_none = False
+
+    # Normalize chunks' outer tuple:
+    chunks_tuple: T_ChunkVals | T_ChunksVals
+    if isinstance(chunks, tuple):
+        chunks_tuple = chunks
+    elif isinstance(chunks, (int, float, str)):
+        chunks_tuple = (chunks,) * shape_len
+    elif isinstance(chunks, dict):
+        chunks_tuple = tuple(chunks.get(i, -1) for i in range(shape_len))
+    elif isinstance(chunks, np.ndarray):
+        chunks_tuple = tuple(chunks.tolist())
+    else:
+        if isinstance(chunks, list):
+            chunks_tuple = tuple(chunks)
+        elif isinstance(chunks, np.ndarray):
+            chunks_tuple = tuple(chunks.tolist())
+        elif chunks is None:
+            raise ValueError(CHUNKS_NONE_ERROR_MESSAGE)
+        else:
+            raise ValueError("Type of chunk is not supported.")
```

Comment:
Maybe this error could be more useful to users if you include the erroneous type in the error message, e.g.  `raise ValueError(f"Type of chunk: {type(chunks)}  is not supported.")`

Reply:
Done, I added the chunks values too.